src/client/websocket_client.js

```diff
@@ -172,35 +174,35 @@ class WebSocketClient {
                     if (msg.error) {
                         console.warn(msg); //eslint-disable-line no-console
                     }
-                } else if (this.eventCallback) {
-                    this.eventCallback(msg, this.dispatch, this.getState);
+                } else if (this.eventAction) {
+                    dispatch(this.eventAction(msg));
                 }
             };
         });
     }
 
-    setConnectingCallback(callback) {
-        this.connectingCallback = callback;
+    setConnectingAction(action) {
```

Comment:
This is going to break anyone else using the websocket client. I'm pretty sure we want backwards compatibility here

Reply:
this is going to break anyway as we changed the way the previous callbacks now actions are called

Reply:
But it should not be a problem as in reality people should be using the WS actions and not the client directly